### PR TITLE
Extract useful components

### PR DESCRIPTION
**Pull request by JamesGlover**
I've added some components, and performed a bit of refactoring as part of the receptions work.

I've extracted this out so @seenanair can use it, and will close and then rebase the rest of my PR.

### CODE REVIEW

File: src/components/shared/TractionMenuItem.vue [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      active ? 'text-white bg-sp' : 'text-sp',
+    ]"
+  >
+    <slot />
+  </li>
+</template>
+
+<script>
+export default {
+  name: 'TractionMenu',
```
Comment (seenanair) [inline src/components/shared/TractionMenuItem.vue:16]: A doubt - Is it not 'TractionMenuItem'?

Comment (JamesGlover) [reply src/components/shared/TractionMenuItem.vue:16]: You are correct. One of the flaws of using a previous component as a template.

Comment (JamesGlover) [reply src/components/shared/TractionMenuItem.vue]: Fixed.

**Comment by JamesGlover** [comment]
Documentation wise it might be worth having a consideration of storybook/Vue Styleguidist to make sure that whatever format we adopt works well with that. But yes, agreed.

**Pull request merged**